src/rgw/rgw_rest_sts.cc

```diff
@@ -187,7 +187,12 @@ int RGWSTSGetSessionToken::get_params()
   tokenCode = s->info.args.get("TokenCode");
 
   if (! duration.empty()) {
-    uint64_t duration_in_secs = stoull(duration);
+    uint64_t duration_in_secs;
+    try {
+      duration_in_secs = stoull(duration);
+    } catch (...) {
+      return -EINVAL;
+    }
```

Comment:
this error handling still doesn't verify that we parsed the whole string, so it would accept stuff like `123abc`

the helper functions in common/strtol.h like `strict_strtoll()` do this extra check, so i'd prefer to use that for duration parsing